### DISCUSSION TITLE
How to create and use empty subdirectory?

### DISCUSSION THREAD

**Discussion by Richard-ITR** [nlohmann/json#3486] [Q&A] [answered]
I created my basic json structure by this initialization:

`
	jobj = nlohmann::ordered_json{
		{ "type", _type },
		{ "version", _version },
		{ "backup", {
			{ "information",{
				{ "general",{
				}
				},
				{ "customer",{
				}
				}
			}
			},
....
`
at this point general and customer are empty. 
later on I want to add programmatically key value pairs:

`
		nlohmann::ordered_json::reference ref = jobj.at("/backup/information/general"_json_pointer);
		if (!ref.is_object()) throw("wrong type");
		ref.push_back({ _key,_value });
`

but **ref** is not valid!

if I would initialize the structures with dummy values, e.g.
`				{ "general",{
					{ "dummy", "" }
				}
				},
`
 at initialization, I am able to add elements with push_back(); but this dummy entry is just for testing, not for final 

How to solve?

**Comment by falbrechtskirchinger** [comment, marked as answer]
The `{}` in `{ "general", {}}` does not construct an object but a `null` JSON value. You might try:
```cpp
if(ref.is_null()) ref = nlohmann::ordered_json::object();
```
Alternatively you can add the empty object at construction using `{ "general", nlohmann::ordered_json::object()}`.

Edit: Remember that `nlohmann::ordered_json{}` likewise constructs a `null` value and not an empty object.

**Reply by Richard-ITR** [reply]
many thanks; nlohmann::ordered_json::object()  works!
